### PR TITLE
Schema scope

### PR DESCRIPTION
**Pull request by neilpeterson**
Updated deployment and parameter snippet schemas:

- Added snippet for Subscription, Management Group, and Tenant deployments
- Updated Resource Group deployment schema version to 2019-04-01
- Updated parameter schema version to 2019-04-01
- Renamed existing snippets (prefix) for clarity
- Updated existing descriptions for clarity

Sample:

image

Fixes https://github.com/microsoft/vscode-azurearmtools/issues/449

### CODE REVIEW

**Comment by nilshedstrom** [comment]
I like that you can create a new arm template with "arm!" and a new parameter file with "armp!"
I would prefer to stick with the current ones compared to "arm-deploy-file-rg"

**Comment by neilpeterson** [comment]
Thanks @nilshedstrom I also agree that the muscle memory / quick keystroke on `arm!` is nice, but it's not very descriptive, particularly after adding the additional scopes. Do you have any thoughts on how to handle when adding the new scopes (subscription, management group, and tenant)?

File: assets/jsonsnippets.jsonc [hunk 1]
```diff
@@ -12,18 +12,63 @@
...
+            "    \"\\$schema\": \"https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#\",",
+            "    \"contentVersion\": \"1.0.0.0\",",
+            "    \"parameters\": {},",
+            "    \"variables\": {},",
+            "    \"resources\": [],",
+            "    \"outputs\": {},",
+            "    \"functions\": []",
+            "}"
+        ],
+        "description": "Tanant Deployment"
```
Comment (StephenWeatherford) [inline assets/jsonsnippets.jsonc]: tenant

**Comment by StephenWeatherford** [comment]
Here's what it looks like with your current changes:
image
Here are a couple other possibilities:
image
image

**Comment by nilshedstrom** [comment]
> image
@neilpeterson I like the last suggestion from @StephenWeatherford

File: assets/jsonsnippets.jsonc (hunk 1 above)
Comment (neilpeterson) [reply assets/jsonsnippets.jsonc]: Thanks, addressed.

**Comment by neilpeterson** [comment]
Thanks for the suggestions @StephenWeatherford  and input @nilshedstrom. Let me gather some additional opinions and I will update.

**Comment by neilpeterson** [comment]
@nilshedstrom / @StephenWeatherford option three wins :).

**Pull request merged**